lib/Doctrine/DBAL/Platforms/OraclePlatform.php

```diff
@@ -975,24 +975,30 @@ public function getName()
      */
     protected function doModifyLimitQuery($query, $limit, $offset = null)
     {
-        $limit = (int) $limit;
```

Comment:
@morozov yeah good point. But then leave it as before casting both parameters to `int` and checking for `0` instead.